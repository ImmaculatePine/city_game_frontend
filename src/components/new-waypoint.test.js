import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import NewWaypoint from './new-waypoint'

describe('NewWaypoint', () => {
  const notUsedPlace = {
    id: 1,
    name: 'Not used place',
    address: 'Not used address'
  }
  const usedPlace = { id: 2, name: 'Used place', address: 'Used address' }
  const game = {
    id: 1,
    name: 'Game 1',
    waypoints: [
      {
        id: 1,
        position: 0,
        place: usedPlace
      }
    ]
  }
  const places = [notUsedPlace, usedPlace]
  const onSubmit = () => {}

  it('renders a list of places which are not used as waypoints yet', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NewWaypoint game={game} places={places} onSubmit={onSubmit} />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders empty div when game is not found', () => {
    const tree = renderer
      .create(
        <NewWaypoint game={undefined} places={places} onSubmit={onSubmit} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
