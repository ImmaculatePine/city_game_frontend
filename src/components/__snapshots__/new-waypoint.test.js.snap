// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewWaypoint renders a list of places which are not used as waypoints yet 1`] = `
<div
  className="modal is-active"
>
  <div
    className="modal-background"
  />
  <div
    className="modal-card"
  >
    <form
      onSubmit={[Function]}
    >
      <header
        className="modal-card-head"
      >
        <p
          className="modal-card-title"
        >
          Add waypoint
        </p>
      </header>
      <section
        className="modal-card-body"
      >
        <div
          className="field"
        >
          <div
            className="control"
          >
            <select
              name="placeId"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              />
              <option
                value={1}
              >
                Not used place
              </option>
            </select>
          </div>
        </div>
      </section>
      <footer
        className="modal-card-foot"
      >
        <button
          className="button is-success"
          onClick={[Function]}
        >
          Add
        </button>
        <a
          className="button"
          href="/games/1"
          onClick={[Function]}
        >
          Back
        </a>
      </footer>
    </form>
  </div>
  <a
    aria-label="close"
    className="modal-close is-large"
    href="/games/1"
    onClick={[Function]}
  />
</div>
`;

exports[`NewWaypoint renders empty div when game is not found 1`] = `<div />`;
